cpp/simulations/abm.cpp

```diff
@@ -17,14 +17,20 @@
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
+#define _CRT_SECURE_NO_DEPRECATE
```

Comment:
It was used to avoid the errors due to the usage of fopen with the new Window compiler. I've switched to use the provided IOResults so this line is now removed.